File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        // N마리 폰켓몬 중에서 N/2마리 선택 -> 종류가 많도록 선택
        
        Map<Integer, Integer> map = new HashMap<>();
        for(int num:nums) {
            map.put(num, map.getOrDefault(num, 0)+1);
        }
        
        return Math.min(map.size(), nums.length/2);
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <unordered_set>
#include <algorithm>
using namespace std;

int solution(vector<int> nums)
{
    unordered_set<int> unique_types(nums.begin(), nums.end());
    return min(nums.size()/2, unique_types.size());
}